harambe/parser/parser.py

```diff
@@ -159,6 +165,25 @@ def _get_type(self, field: str) -> Type[Any]:
             raise ValueError(f"Unsupported field type: {field}")
         return field_type
 
+    def _all_fields_empty(self, data: dict[str, Any]) -> bool:
+        """
+        Recursively check if all fields in the data are either None or empty.
+        This includes handling nested dictionaries and lists.
+        """
+
+        def is_empty(value: Any) -> bool:
+            if value is None:
+                return True
+            if isinstance(value, (str, list, dict)):
+                if isinstance(value, dict):
+                    return all(is_empty(v) for v in value.values())
+                if isinstance(value, list):
+                    return all(is_empty(v) for v in value)
+                return len(value) == 0
```

Comment:
some needless nesting here

```suggestion
             if isinstance(value, dict):
                 return all(is_empty(v) for v in value.values())
             if isinstance(value, list):
                 return all(is_empty(v) for v in value)
             if isinstance(value, str)
              return bool(value)
```